beater/handlers.go

```diff
@@ -108,12 +111,40 @@ func healthCheckHandler(_ ProcessorFactory, _ Config, _ reporter) http.Handler {
 		}))
 }
 
+type logContextKey string
+
+var reqLoggerContextKey = logContextKey("requestLogger")
+
 func logHandler(h http.Handler) http.Handler {
+	logger := logp.NewLogger("request")
+
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		logp.Debug("handler", "Request: URI=%s, method=%s, content-length=%d", r.RequestURI, r.Method, r.ContentLength)
+		reqID := uuid.NewV4()
+
 		requestCounter.Inc()
-		h.ServeHTTP(w, r)
+
+		reqLogger := logger.With("request_id", reqID)
+
+		lr := r.WithContext(
+			context.WithValue(r.Context(), reqLoggerContextKey, reqLogger),
+		)
+		lr.Context().Value(reqLoggerContextKey)
+
+		lw := utility.NewRecordingResponseWriter(w)
+
+		h.ServeHTTP(lw, lr)
+
+		reqLogger.Infow("handled request", "response_code", lw.Code,
+			"method", r.Method, "URL", r.URL, "content_length", r.ContentLength,
+			"remote_address", extractIP(r), "user_agent", r.Header.Get("User-Agent"))
+
+		if lw.Code > 399 {
```

Comment:
Should this be the other way around? `error if Code > 399`

Reply:
fixed, thanks! 🤦‍♂️